lib/dal/tests/integration_test/schema/variant/authoring/regenerate.rs

```diff
@@ -89,6 +91,145 @@ async fn regenerate_variant(ctx: &mut DalContext) {
     assert!(funcs_for_default.into_iter().any(|func| func.id == func_id));
 }
 
+#[test]
+async fn update_socket_data_on_regenerate(ctx: &mut DalContext) {
+    let name = "Bandit";
+    let description = None;
+    let link = None;
+    let category = "Blue Heelers";
+    let color = "#00A19B";
+
+    // Create an asset with a corresponding asset func. After that, commit.
+    let schema_variant_id = {
+        let schema_variant = VariantAuthoringClient::create_schema_and_variant(
+            ctx,
+            name,
+            description.clone(),
+            link.clone(),
+            category,
+            color,
+        )
+        .await
+        .expect("unable to create schema and variant");
+        schema_variant.id()
+    };
+    let asset_func = "function main() {
+        const asset = new AssetBuilder();
+        const beta_destination_output_socket = new SocketDefinitionBuilder()
+            .setName(\"input_socket\")
+            .setArity(\"one\")
+            .build();
+        asset.addInputSocket(beta_destination_output_socket);
+
+        return asset.build();
+    }";
+    VariantAuthoringClient::save_variant_content(
+        ctx,
+        schema_variant_id,
+        name,
+        name,
+        category,
+        description.clone(),
+        link.clone(),
+        color,
+        ComponentType::Component,
+        Some(asset_func),
+    )
+    .await
+    .expect("could not save content");
+    ChangeSetTestHelpers::commit_and_update_snapshot_to_visibility(ctx)
+        .await
+        .expect("could not commit");
+
+    // Once it's all ready, regenerate and commit.
+    VariantAuthoringClient::regenerate_variant(ctx, schema_variant_id)
+        .await
+        .expect("could not regenerate variant");
+    ChangeSetTestHelpers::commit_and_update_snapshot_to_visibility(ctx)
+        .await
+        .expect("could not commit");
+
+    // create a component
+    let component =
+        create_component_for_default_schema_name_in_default_view(ctx, "Bandit", "Bluey")
+            .await
+            .expect("could not create component");
+    let component_id = component.id();
+    ChangeSetTestHelpers::commit_and_update_snapshot_to_visibility(ctx)
+        .await
+        .expect("could not commit");
+
+    // check the input socket arity
+    let mut input_socket_avs = component
+        .input_socket_attribute_values(ctx)
+        .await
+        .expect("could not get input sockets");
+    assert!(input_socket_avs.len() == 1);
```

Comment:
Thank you! The whole "list call but only want one" pattern is annoying. Good way to handle it!